gammapy/spectrum/energy_group.py

```diff
@@ -230,110 +231,95 @@ def find_list_idx(self, energy):
-
-    def reindex_groups(self):
-        """Re-index groups"""
-        for energy_group_idx, group in enumerate(self):
-            group.energy_group_idx = energy_group_idx
-
-    def make_merged_group(self, list_idx_min, list_idx_max, bin_type):
-        """Merge group according to indexes"""
-        left_group = self[list_idx_min]
-        right_group = self[list_idx_max]
-
-        return SpectrumEnergyGroup(
-            energy_group_idx=INVALID_GROUP_INDEX,
-            bin_idx_min=left_group.bin_idx_min,
-            bin_idx_max=right_group.bin_idx_max,
-            bin_type=bin_type,
-            energy_min=left_group.energy_min,
-            energy_max=right_group.energy_max,
-        )
-
-    # TODO: choose one of the apply energy min / max methods!
-
-    def apply_energy_min_old(self, energy):
-        """Modify list in-place to apply a min energy cut."""
-        idx_max = self.find_list_idx(energy)
-        self.make_and_replace_merged_group(0, idx_max, 'underflow')
-
-    def apply_energy_min(self, energy):
-        t = self.to_group_table()
-        idx_max = np.where(t['energy_min'].quantity < energy)[0][-1]
-        self.make_and_replace_merged_group(0, idx_max, 'underflow')
-
-    def apply_energy_max_old(self, energy):
-        """Modify list in-place to apply a max energy cut."""
-        idx_min = self.find_list_idx(energy)
-        idx_max = len(self) - 1
-        self.make_and_replace_merged_group(idx_min, idx_max, 'overflow')
-
-    def apply_energy_max(self, energy):
-        t = self.to_group_table()
-        idx_min = np.where(t['energy_max'].quantity > energy)[0][0]
-        self.make_and_replace_merged_group(idx_min, len(self) - 1, 'overflow')
-
-    def clip_to_valid_range(self, list_idx):
-        """TODO: document"""
-        if self[list_idx].bin_type == 'underflow':
-            list_idx += 1
-
-        if self[list_idx].bin_type == 'overflow':
-            list_idx -= 1
-
-        if list_idx < 0:
-            raise IndexError('list_idx {} < 0'.format(list_idx))
-        if list_idx >= len(self):
-            raise IndexError('list_idx {} > len(self)'.format(list_idx))
-
-        return list_idx
-
-    def apply_energy_binning(self, ebounds):
-        """Apply an energy binning."""
-
-        for idx in range(len(ebounds) - 1):
-            energy_min = ebounds[idx]
-            energy_max = ebounds[idx + 1]
-            list_idx_min, list_idx_max = self.find_list_idx_range(energy_min, energy_max)
-
-            # Be sure to leave underflow and overflow bins alone
-            # TODO: this is pretty ugly ... make it better somehow!
-            list_idx_min = self.clip_to_valid_range(list_idx_min)
-            list_idx_max = self.clip_to_valid_range(list_idx_max)
-
-            self.make_and_replace_merged_group(
-                list_idx_min=list_idx_min,
-                list_idx_max=list_idx_max,
-                bin_type='normal',
-            )
 
 
 class SpectrumEnergyGroupMaker(object):
```

Comment:
I don't remember what we agreed on here in the end. Didn't we say than this Maker should be removed and the algorithm added as classmethod on the container object?